### PR TITLE
Various logging improvements

### PR DESCRIPTION
**Pull request by uce**
I would like to include the following fixes for the 1.1.4 release. In general, I try to improve the debugging experience with this.

1. **Reduce log pollution**: Some debug logs were very noisy and actually dominate the logs although they provide little value.
    - Log heartbeats on TRACE level
    - Don't log InputChannelDeploymentDescriptor
    - Decrease HadoopFileSystem logging
2. **Improve log messages**: Some existing debug log messages were not that helpful.
    - Log GlobalConfiguration loaded properties on INFO level
    - Add TaskState toString
    - Add more detailed log messages to HA job graph store
3. **Improve existing logger configuration templates**: The existing template simply configured the appenders and left everything except the root logger to the user. I changed it to be more fine grained (root logger, Flink, common libs/connectors). The goal is that users trying to DEBUG Flink, don't end up with too many unrelated log messages. 

/cc @tillrohrmann

### CODE REVIEW

File: flink-dist/src/main/flink-bin/conf/log4j.properties [hunk 1]
```diff
@@ -16,14 +16,26 @@
...
 
+# Uncomment this if you want to _only_ change Flink's logging
+#log4j.logger.org.apache.flink=INFO
+
+# The following lines keep the log level of common libraries/connectors on
+# log level INFO. The root logger does not override this. You have to manually
+# change the log levels here.
+log4j.logger.akka=INFO
+log4j.logger.org.apache.kafka=INFO
+log4j.logger.org.apache.hadoop=INFO
```
Comment (tillrohrmann) [inline flink-dist/src/main/flink-bin/conf/log4j.properties]: Does this also work for the shaded hadoop dependencies?

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/TaskState.java [hunk 2]
```diff
@@ -168,4 +168,21 @@ public boolean equals(Object obj) {
 	public int hashCode() {
 		return parallelism + 31 * Objects.hash(jobVertexID, subtaskStates, kvStates);
 	}
+
+	@Override
+	public String toString() {
+		long stateSize = 0L;
+		for (SubtaskState subtaskState : subtaskStates.values()) {
+			stateSize += subtaskState.getStateSize();
+		}
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/TaskState.java]: can't we use the `getStateSize` method here?

File: flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java [hunk 3]
```diff
@@ -211,17 +209,14 @@ public HadoopFileSystem(Class<? extends org.apache.hadoop.fs.FileSystem> fsClass
 				if (new File(possibleHadoopConfPath).exists()) {
 					if (new File(possibleHadoopConfPath + "/core-site.xml").exists()) {
 						retConf.addResource(new org.apache.hadoop.fs.Path(possibleHadoopConfPath + "/core-site.xml"));
-
-						if (LOG.isDebugEnabled()) {
-							LOG.debug("Adding " + possibleHadoopConfPath + "/core-site.xml to hadoop configuration");
-						}
+					} else {
+						LOG.debug("File " + possibleHadoopConfPath + "/core-site.xml not found.");
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java]: `{}` for variables.

File: flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java [hunk 4]
```diff
@@ -211,17 +209,14 @@ public HadoopFileSystem(Class<? extends org.apache.hadoop.fs.FileSystem> fsClass
...
 					}
+
 					if (new File(possibleHadoopConfPath + "/hdfs-site.xml").exists()) {
 						retConf.addResource(new org.apache.hadoop.fs.Path(possibleHadoopConfPath + "/hdfs-site.xml"));
-
-						if (LOG.isDebugEnabled()) {
-							LOG.debug("Adding " + possibleHadoopConfPath + "/hdfs-site.xml to hadoop configuration");
-						}
+					} else {
+						LOG.debug("File " + possibleHadoopConfPath + "/hdfs-site.xml not found.");
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java]: Same here with `{}`.

File: flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/TaskState.java (hunk 2 above)
Comment (uce) [inline flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/TaskState.java]: Yes

File: flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java (hunk 4 above)
Comment (uce) [inline flink-runtime/src/main/java/org/apache/flink/runtime/fs/hdfs/HadoopFileSystem.java]: Sure

File: flink-dist/src/main/flink-bin/conf/log4j.properties (hunk 1 above)
Comment (uce) [inline flink-dist/src/main/flink-bin/conf/log4j.properties]: No, at runtime they have a different package. Therefore, we need to change this here, good catch I think ;) @rmetzger Can you confirm?

Comment (uce) [inline flink-dist/src/main/flink-bin/conf/log4j.properties]: Ah, now I remember ;) This is indeed shady business. But good to know that we can keep it as is.

**Comment by tillrohrmann** [comment]
We should also set the received and handled message in `FlinkUntypedActor#onReceive` to trace.

**Comment by uce** [comment]
Added `FlinkUntypedActor` as well and addressed the comments, Till. I would like to merge this now.

**Pull request closed without merging**